plugins/inputs/ctrlx_datalayer/ctrlx_datalayer.go

```diff
@@ -0,0 +1,363 @@
+	request := sub.createRequest(id)
+	payload, err := json.Marshal(request)
+	if err != nil {
+		return "", fmt.Errorf("failed to create subscription %d payload, error: %w", sub.index, err)
+	}
+
+	requestBody := bytes.NewBuffer(payload)
+	req, err := http.NewRequest("POST", sseURL, requestBody)
+	if err != nil {
+		return "", fmt.Errorf("failed to create subscription %d request, error: %w", sub.index, err)
+	}
+
+	req.Header.Add("Authorization", c.tokenManager.Token.String())
+
+	resp, err := c.connection.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("failed to do request to create sse subscription %d, error: %w", sub.index, err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 && resp.StatusCode != 201 {
+		return "", fmt.Errorf("failed to create sse subscription %d, status: %q", sub.index, resp.Status)
+	}
+
+	return sseURL + "/" + id, nil
+}
+
+// createSubscriptionAndSseClient creates a sse subscription on the server and
+// initializes a sse client to receive sse events from the server.
+func (c *CtrlXDataLayer) createSubscriptionAndSseClient(sub *Subscription) (*sseclient.SseClient, error) {
+	t, err := c.tokenManager.RequestAuthToken()
+	if err != nil {
+		return nil, err
+	}
+
+	subURL, err := c.createSubscription(sub)
+	if err != nil {
+		return nil, err
+	}
+
+	client := sseclient.NewSseClient(subURL, t.String(), c.InsecureSkipVerify)
+
+	return client, nil
+}
+
+// addMetric writes sse metric into accumulator.
+func (c *CtrlXDataLayer) addMetric(se *sseclient.SseEvent, sub *Subscription) {
+	switch se.Event {
+	case "update":
+		// Received an updated value, that we translate into a metric
+		var d sseEventData
+
+		if err := json.Unmarshal([]byte(se.Data), &d); err != nil {
+			c.acc.AddError(fmt.Errorf("received malformed data from 'update' event"))
+			return
+		}
+		m, err := c.createMetric(&d, sub)
+		if err != nil {
+			c.acc.AddError(fmt.Errorf("failed to create metrics: %v", m))
+			return
+		}
+		c.acc.AddMetric(m)
+	case "error":
+		// Received an error event, that we report to the accumulator
+		var e sseEventError
+		if err := json.Unmarshal([]byte(se.Data), &e); err != nil {
+			c.acc.AddError(fmt.Errorf("received malformed data from 'error' event: %w", err))
+			return
+		}
+		c.acc.AddError(fmt.Errorf("received 'error' event for node: %q", e.Instance))
+	case "keepalive":
+		// Keepalive events are ignored for the moment
+		c.Log.Debug("Received keepalive event")
+	default:
+		// Received a yet unsupported event type
+		c.Log.Debug("Received unsupported event: %q", se.Event)
+	}
+}
+
+// createMetric - create metric depending on flag 'output_json' and data type
+func (c *CtrlXDataLayer) createMetric(em *sseEventData, sub *Subscription) (telegraf.Metric, error) {
+	t := convertTimestamp2UnixTime(em.Timestamp)
+	node := sub.node(em.Node)
+	if node == nil {
+		return nil, errors.New("node not found")
+	}
+
+	// default tags
+	tags := map[string]string{
+		"node":       em.Node,
```

Comment:
Can you please name the tag `source` as this is what we are currently do in other plugins in order to unify that over time.

Reply:
**Node** is the term used by our users to identify an object in the data layer. **Source** would be to generic and is used in a different way in our system.